backend/sites/src/app/resolvers/landHistory/landHistory.resolver.ts

```diff
@@ -0,0 +1,55 @@
+import { Args, Query, Resolver } from '@nestjs/graphql';
+import { RoleMatchingMode, Roles } from 'nest-keycloak-connect';
+import { GenericResponseProvider } from '../../dto/response/genericResponseProvider';
+import { LandHistories } from '../../entities/landHistories.entity';
+import { LandHistoryResponse } from '../../dto/landHistory.dto';
+import { LandHistoryService } from '../../services/landHistory/landHistory.service';
+
+type SortDirection = 'ASC' | 'DESC';
+
+@Resolver(() => LandHistories)
+export class LandHistoryResolver {
+  constructor(
+    private readonly landHistoryService: LandHistoryService,
+    private readonly genericResponseProvider: GenericResponseProvider<
+      LandHistories[]
+    >,
+  ) {}
+
+  @Roles({ roles: ['site-admin'], mode: RoleMatchingMode.ANY })
+  @Query(() => LandHistoryResponse, { name: 'getLandHistoriesForSite' })
+  // TODO: figure out how this works and if we need GenericValidationPipe here
+  // Passing an empty string to a nullable field throws a validation error
+  // @UsePipes(new GenericValidationPipe())
+  async getLandHistoriesForSite(
+    @Args('siteId', { type: () => String })
+    siteId: string,
+
+    @Args('searchTerm', { type: () => String, nullable: true })
+    searchTerm: string,
+
+    @Args('sortDirection', { nullable: true })
+    sortDirection: SortDirection,
+  ) {
+    const result = await this.landHistoryService.getLandHistoriesForSite(
+      siteId,
+      searchTerm,
+      sortDirection,
+    );
+    if (result.length > 0) {
```

Comment:
should you check if result is empty before performing result.length ?

Reply:
`getLandHistoriesForSite` calls typeorm's `getMany`, which should always return an array, so I think we're safe here.